packages/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp

```diff
@@ -262,9 +262,27 @@ jsi::Value ShareableWorklet::toJSValue(jsi::Runtime &rt) {
           data_.cend(),
           [](const auto &item) { return item.first == "__workletHash"; }) &&
       "ShareableWorklet doesn't have `__workletHash` property");
+  // Copy the code & sourceURL from the worklet object to ensure
+  // that the code is not modified in the original object.
   jsi::Value obj = ShareableObject::toJSValue(rt);
+  auto initData = obj.asObject(rt).getProperty(rt, "__initData").asObject(rt);
+  auto code = std::make_shared<const jsi::StringBuffer>(
+      initData.getProperty(rt, "code").asString(rt).utf8(rt));
+  auto sourceURL = initData.getProperty(rt, "location").asString(rt).utf8(rt);
+  // The code has to be evaluated in the context of the worklet runtime.
+  // This is done by creating a new function that evaluates the code and
+  // returns the resulting function.
+  auto evaluateWorkletFunction = jsi::Function::createFromHostFunction(
```

Comment:
this should be made shared, no?

Reply:
No, it's executed during the call to valueUnpacker and is allowed to be destroyed afterwards.
The worklet functions that it returns is retained by the runtime in the worklets cache so that doesn't get destroyed.